x/wasm/keeper/keeper.go

```diff
@@ -125,11 +132,14 @@ func NewKeeper(
 		queryGasLimit:    wasmConfig.SmartQueryGasLimit,
 		paramSpace:       paramSpace,
 	}
-	keeper.wasmVMQueryHandler = DefaultQueryPlugins(bankKeeper, stakingKeeper, distKeeper, channelKeeper, queryRouter, &keeper)
+
+	keeper.wasmVMQueryHandler = DefaultQueryPlugins(bankKeeper, stakingKeeper, distKeeper, channelKeeper, queryRouter, keeper)
 	for _, o := range opts {
-		o.apply(&keeper)
+		o.apply(keeper)
 	}
-	return keeper
+	// not updateable, yet
+	keeper.wasmVMResponseHandler = NewDefaultWasmVMContractResponseHandler(NewMessageDispatcher(keeper.messenger, keeper))
```

Comment:
IMHO it makes perfect sense to have this as an extension point that can be set from outside so that people can implement their own response handling strategy. This can be very powerful but this would be incompatible with other `Options` we have already. I kept this out of this PR.